core/src/inputs/stack.rs

```diff
@@ -65,3 +65,10 @@ impl IntoIterator for StackInputs {
         self.values.into_iter()
     }
 }
+
+impl Serializable for StackInputs {
+    fn write_into<W: ByteWriter>(&self, target: &mut W) {
+        target.write_u64(self.values.len() as u64);
+        self.values.iter().copied().for_each(|v| target.write(v));
```

Comment:
The length will never be greater than `u32` and probably will never be greater than `u16` either. We could go with `u16` here for length - but then we'd need to panic if for w/e reason it is greater. So, maybe for now lets leave a TODO to address this in the future.

Reply:
Changed to `u32` and added a debug assert with a TODO comment to tweak in the future